kalite/topic_tools/annotate.py

```diff
@@ -50,97 +50,71 @@ def update_content_availability(content_list, language="en"):
                     else:
                         subtitle_langs[filename] = [lc]
 
-    # English-language exercises live in application space, translations in user space
-
-    if language == "en":
-        exercise_root = os.path.join(settings.KHAN_EXERCISES_DIRPATH, "exercises")
-    else:
-        exercise_root = i18n.get_localized_exercise_dirpath(language)
-    if os.path.exists(exercise_root):
-        try:
-            exercise_templates = os.listdir(exercise_root)
-        except OSError:
-            exercise_templates = []
-    else:
-        exercise_templates = []
-
     for content in content_list:
         # Some nodes are duplicated, but they require the same information
         # regardless of where they appear in the topic tree
+
         update = {}
 
         if content.get("kind") == "Exercise":
 
-            # The central server doesn't have an assessment item database
-            if django_settings.CENTRAL_SERVER:
-                continue
-            elif content.get("uses_assessment_items", False):
-                items = []
-                assessment_items = content.get("all_assessment_items", [])
-                for item in assessment_items:
-                    item = json.loads(item)
-                    if get_assessment_item_data(request=None, assessment_item_id=item.get("id")):
-                        items.append(item)
-                        update["available"] = True
-                update["all_assessment_items"] = items
-            else:
-                exercise_file = content.get("name", "") + ".html"
-
-                if language == "en":
-                    exercise_template = exercise_file
-                    if exercise_template in exercise_templates:
-                        update["available"] = True
-                        update["template"] = exercise_template
-                else:
-                    exercise_template = os.path.join(language, exercise_file)
-                    if exercise_template in exercise_templates:
-                        update["available"] = True
-                        update["template"] = exercise_template
+            # Databases have been pre-filtered to only contain existing exercises
+            if database_exists(language=language, channel=channel):
```

Comment:
On further thought, make this contingent on the existence of assessment_items.version - we are going to assume that if you have downloaded assessment items zip, then all exercises that can be available are available (ones not available in that language will not be in the content db at all).